### ISSUE TITLE
One or more errors occurred. (Arithmetic operation resulted in an overflow.)

### ISSUE THREAD

**Issue by shimizu-izumi** [4sval/FModel#314]
### Game

Tower of Fantasy

### Error

```shell
[ERR] One or more errors occurred. (Arithmetic operation resulted in an overflow.)
      at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
```

### Reproduction steps

1. Load Tower of Fantasy
2. Enter the AES Key (0x6E6B325B02B821BD46AF6B62B1E929DC89957DC6F8AA78210D5316798B7508F8)
3. Select GAME_UE4_26 as the UE Version in the settings
4. Open any file that isn't a .locres, .ini, .uplugin or .uproject

**Comment by shimizu-izumi** [comment]
> [https://i.ibb.co/2F92CWH/Screenshot-2022-08-22-194919.png](url) same happened too for CN Version TOF :(

I fixed this issue myself. I needed to patch how CUE4Parse loads the uasset files.

**Comment by Angoks** [comment]
> > [https://i.ibb.co/2F92CWH/Screenshot-2022-08-22-194919.png](url) same happened too for CN Version TOF :(
> 
> I fixed this issue myself. I needed to patch how CUE4Parse loads the uasset files.

oh yeah could you help me with that? maybe i could send message on your twitter or anywhere?

**Comment by GMatrixGames** [comment]
> > [https://i.ibb.co/2F92CWH/Screenshot-2022-08-22-194919.png](url) same happened too for CN Version TOF :(
> 
> I fixed this issue myself. I needed to patch how CUE4Parse loads the uasset files.

Feel free to make a fork and PR it so it can be implemented into upstream CUE4Parse. We'd rather not end up like UEViewer with all game-specific fixes being gatekept 👍

**Comment by 0xDema** [comment]
So are there any steps to get this workaround working?

**Comment by koiseka** [comment]
Wondering the same thing

**Comment by IncinerateZ** [comment]
For anyone still having this issue apply the changes to `FPackageFileSummary.cs` from
[https://github.com/FabianFG/CUE4Parse/commit/19dfffa9e55d0951e7da75789e65c8c4e6e62a3e](https://github.com/FabianFG/CUE4Parse/commit/19dfffa9e55d0951e7da75789e65c8c4e6e62a3e)

**Comment by shimizu-izumi** [comment]
> 

These changes are already in the official FModel version.

**Comment by IncinerateZ** [comment]
> > For anyone still having this issue apply the changes to FPackageFileSummary.cs from
> > [FabianFG/CUE4Parse@19dfffa](https://github.com/FabianFG/CUE4Parse/commit/19dfffa9e55d0951e7da75789e65c8c4e6e62a3e)
> 
> These changes are already in the official FModel version.

Huh weird then cuz i just cloned the repo today and i had this issue

**Comment by shimizu-izumi** [comment]
> > > For anyone still having this issue apply the changes to FPackageFileSummary.cs from
> > > [FabianFG/CUE4Parse@19dfffa](https://github.com/FabianFG/CUE4Parse/commit/19dfffa9e55d0951e7da75789e65c8c4e6e62a3e)
> > 
> > 
> > These changes are already in the official FModel version.
> 
> Huh weird then cuz i just cloned the repo today and i had this issue

Yeah because these changes are only on the dev branch, but the current release build is made with the dev branch.

**Comment by 4sval** [comment]
released
